server/api.js

```diff
@@ -9,29 +9,107 @@ router.get("/", (req, res) => {
 
 router.get("/clients", (req, res) => {
 	db.query("SELECT * from clients;")
-    .then((result) => res.json(result.rows))
-	.catch((e) => {
-		console.error(e);
-		res.sendStatus(400);
-	});
+		.then((result) => res.json(result.rows))
+		.catch((e) => {
+			res.status(400).send(e);
+		});
 });
 
-router.post("/clients", (req, res) => {
-
-	const newClientDateAdded = req.body.date_added;
-	const newClientName = req.body.name;
-	const newClientBikes = req.body.bikes_needed;
-	const newClientPhoneNumber = req.body.phone_number;
+const saveUser = async (element) => {
+	return new Promise((resolve, reject) => {
+		const newClientDateAdded = element.date_added;
+		if (!newClientDateAdded || isNaN(!newClientDateAdded)) {
```

Comment:
This condition is always going to be true, because newClientDateAdded is just a string. 
`isNaN(!newClientDateAdded)` means `!newClientDateAdded` is always going to be false, and `isNaN(false)` will always be true. This causes the server to reject all my POST requests.
The `isNaN` check should be moved below, since `Date.parse(newClientDateAdded)` can actually return NaN.